torch/csrc/jit/script/compiler.cpp

```diff
@@ -507,11 +524,30 @@ at::optional<std::vector<Value*>> tryMatchSchema(
     std::vector<Value*> positional_inputs;
     std::vector<bool> used_kwarg(kwargs.size(), false);
 
-    for(size_t i = 0; i < schema.arguments.size(); ++i) {
-      const auto& arg = schema.arguments[i];
+    // if we finish the loop will we have consumed all arguments?
+    size_t used_args = 0;
+
+    for(size_t schema_i = 0; schema_i < schema.arguments.size(); ++schema_i) {
+      const auto& arg = schema.arguments[schema_i];
       at::optional<NamedValue> v;
-      if(i < args.size()) {
-        v = args[i];
+      if(!arg.kwarg_only && schema_i < args.size()) {
+
+        // allow zeros(IntList sizes) to work with zeros(1, 2) or zeros(1)
```

Comment:
It would be really nice if this was cross-referenced with the relevant code in our python argument parser which implements similar behavior in eager mode. I spent some time looking for it in PythonArgs::intlist, but I didn't see the code that would have actually implemented it.

Reply:
In particular, if the condition in the argument parser changes, this condition must change too.

Reply:
Yes, but I can't find it either. @gchanan do you know where the vararg parsing in python arg parser is?